dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemIdentifierControllerIT.java

```diff
@@ -0,0 +1,178 @@
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.identifier.DOI;
+import org.dspace.identifier.DOIIdentifierProvider;
+import org.dspace.identifier.factory.IdentifierServiceFactory;
+import org.dspace.identifier.service.DOIService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test getting and registering item identifiers
+ *
+ * @author Kim Shepherd
+ */
+public class ItemIdentifierControllerIT extends AbstractControllerIntegrationTest {
+    @Before
+    public void setup() throws Exception {
+        super.setUp();
+    }
+
+    @Test
+    public void testRegisterDoiForItem() throws Exception {
+        //Turn off the authorization system, otherwise we can't make the objects
+        context.turnOffAuthorisationSystem();
+
+        DOIService doiService = IdentifierServiceFactory.getInstance().getDOIService();
+
+        //** GIVEN **
+        //1. A community-collection structure with one parent community and two collections.
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+
+        //2. A public item that is readable by Anonymous
+        Item publicItem1 = ItemBuilder.createItem(context, col1)
+                .withTitle("Public item 1")
+                .withIssueDate("2017-10-17")
+                .withAuthor("Smith, Donald")
+                .build();
+
+        // This item should not have a DOI
+        DOIIdentifierProvider doiIdentifierProvider =
+                DSpaceServicesFactory.getInstance().getServiceManager()
+                        .getServiceByName("org.dspace.identifier.DOIIdentifierProvider",
+                                org.dspace.identifier.DOIIdentifierProvider.class);
+        doiIdentifierProvider.delete(context, publicItem1);
+
+        // A non-admin should get an unauthorised error from REST method preauth
+        // Expect first forbidden
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(post("/api/core/items/" +
+                        publicItem1.getID().toString() + "/identifiers"))
+                .andExpect(status().isForbidden());
+
+        // Set token to admin credentials
+        token = getAuthToken(admin.getEmail(), password);
+
+        // Expect a successful 201 CREATED for this item with no DOI
+        getClient(token).perform(post("/api/core/items/" +
+                        publicItem1.getID().toString() + "/identifiers"))
```

Comment:
Maybe I'm reading these tests incorrectly... but is this `POST` sending empty data?  How do we know that means to create a DOI and not a Handle (or "other" identifier)?  The behavior here seems odd, but maybe it's cause I don't have a REST Contract to clarify it.

Reply:
yes, the POST is simply a "create" operation, and while I was working specifically with DOIs there was no data that could be useful to submit -- in some cases we already have the DOI minted, this is just a manual admin action to say "And now, either mint or promote that DOi to 'to be registered' status". In other cases, we also mint a new DOI first. But the number, prefix, etc., should all come from configuration.

However as per your comments and my answers above, I can see that setting this up for future expansion by at least posting an identifier type, might make things a bit easier to follow.